src/dotnet/APIView/APIViewWeb/LeanControllers/ReviewsController.cs

```diff
@@ -0,0 +1,105 @@
+        [HttpGet(Name = "GetReviews")]
+        public async Task<ActionResult<PagedList<ReviewListItemModel>>> GetReviewsAsync([FromQuery] PageParams pageParams, [FromQuery] ReviewFilterAndSortParams filterAndSortParams)
+        {
+            var result = await _reviewManager.GetReviewsAsync(pageParams, filterAndSortParams);
+            Response.AddPaginationHeader(new PaginationHeader(result.NoOfItemsRead, result.PageSize, result.TotalCount));
+            return new LeanJsonResult(result, StatusCodes.Status200OK);
+        }
+
+        /// <summary>
+        /// Endpoint used by Client SPA for creating reviews.
+        /// </summary>
+        /// <param name="reviewCreationParam"></param>
+        /// <returns></returns>
+        [HttpPost(Name = "CreateReview")]
+        public async Task<ActionResult<APIRevisionListItemModel>> CreateReviewAsync([FromForm] ReviewCreationParam reviewCreationParam)
+        {
+            var review = await _reviewManager.GetOrCreateReview(file: reviewCreationParam.File, filePath: reviewCreationParam.FilePath, language: reviewCreationParam.Language);
+
+            if (review != null)
+            {
+                APIRevisionListItemModel apiRevision = await _apiRevisionsManager.CreateAPIRevisionAsync(user: User, review: review, file: reviewCreationParam.File, 
+                    filePath: reviewCreationParam.FilePath, language: reviewCreationParam.Language, label: reviewCreationParam.Label);
+                return new LeanJsonResult(apiRevision, StatusCodes.Status201Created);
+            }
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        ///<summary>
+        ///Retrieve the Content (codeLines and Navigation) of a review
+        ///</summary>
+        ///<param name="reviewId"></param>
+        ///<param name="revisionId"></param>
+        ///<returns></returns>
+        [HttpGet]
+        [Route("{reviewId}/content")]
+        public async Task<ActionResult<ReviewContentModel>> GetReviewContentAsync(string reviewId, [FromQuery]string revisionId=null)
+        {
+           var review = await _reviewManager.GetReviewAsync(user:User, id: reviewId);
+           var revisions = await _apiRevisionsManager.GetAPIRevisionsAsync(reviewId);
+           var activeRevision = (string.IsNullOrEmpty(revisionId)) ? 
+               await _apiRevisionsManager.GetLatestAPIRevisionsAsync(reviewId, revisions) : await _apiRevisionsManager.GetAPIRevisionAsync(user: User, apiRevisionId: revisionId);
+           var comments = await _commentManager.GetReviewCommentsAsync(reviewId);
+
+           var renderableCodeFile = await _codeFileRepository.GetCodeFileAsync(activeRevision.Id, activeRevision.Files[0].FileId);
+           var reviewCodeFile = renderableCodeFile.CodeFile;
+           var fileDiagnostics = reviewCodeFile.Diagnostics ?? Array.Empty<CodeDiagnostic>();
+           var htmlLines = renderableCodeFile.Render(showDocumentation: false);
```

Comment:
Will we need to call this GetReviewContentAsync with documentation? Which API is used to get review with documentation? It's better to add a n optional param for showDocumentation for this API and use it instead if hardcoding false.

Reply:
Sorry, this endpoint should not be in this change. Thats for the review page which I plan on converting to SPA long term but not right now. So I will remove this completely.